doc/tutorials/cox_datafit.rst

```diff
@@ -0,0 +1,227 @@
+
+Datafit expression
+------------------
+
+To get the expression of the Cox datafit, we refer to the expression of the negative log-likelihood according to Breslow estimate [`1`_, Section 2]
+
+.. math::
+    :label: breslow-estimate
+
+    l(\beta) = \frac{1}{n} \sum_{i=1}^{n} -s_i \langle x_i, \beta \rangle + s_i \log(\sum_{y_j \geq y_i} e^{\langle x_j, \beta \rangle})
+    .
+
+
+Ideally, we ought to have a vectorized expression to ease gradient and Hessian derivation.
+
+.. note::
+
+    Having a vectorized expression would enable us to leverage ``Numpy`` broadcasting and vectorization
+    and gain in terms of efficiency.
+
+
+We introduce the matrix :math:`mathbf{B} \in \mathbb{R}^{n \times n}` defined as :math:`\mathbf{B}_{i, j} = \mathbb{1}_{y_j \geq y_i} = 1` if :math:`y_j \geq y_i` and :math:`0` otherwise.
+
+We notice that the first term in the sum can we rewritten as
+
+.. math::
+
+    \sum_{i=1}^{n} -s_i \langle x_i, \beta \rangle = -\langle s, \mathbf{X}\beta \rangle
+    ,
+
+whereas the second term
+
+.. math::
+
+    \sum_{i=1}^n s_i \log(\sum_{y_j \geq y_i} e^{\langle x_j, \beta \rangle}) = \sum_{i=1}^n s_i \log(\sum_{j=1}^n \mathbb{1}_{y_j \geq y_i} e^{\langle x_j, \beta \rangle}) = \langle s, \log(\mathbf{B}e^{\mathbf{X}\beta}) \rangle
+    ,
+
+where the :math:`\log` is performed element-wise. Therefore we deduce the expression of Cox datafit
+
+.. math::
+    :label: vectorized-cox-breslow
+
+    l(\beta) =  -\frac{1}{n} \langle s, \mathbf{X}\beta \rangle + \frac{1}{n} \langle s, \log(\mathbf{B}e^{\mathbf{X}\beta}) \rangle
+    .
+
+We observe from this vectorized reformulation that Cox datafit depends only :math:`\mathbf{X}\beta`. On the one hand, this illustrate that it fits well the GLM framework. On the other, now on, we can focus the simplified function
+
+.. math::
+    :label: simple-function
+
+    F(u) = -\frac{1}{n} \langle s, u \rangle + \frac{1}{n} \langle s, \log(\mathbf{B}e^u) \rangle
+    ,
+
+to derive the gradient and Hessian.
+
+
+Gradient and Hessian
+--------------------
+
+Now that we have a vectorized expression of the datafit, we can apply the chain rule to derive its gradient and Hessian.
+
+For some :math:`u \in \mathbb{R}^n`, The gradient is
+
+.. math::
+    :label: raw-gradient
+
+    \nabla F(u) = -\frac{1}{n} s + \frac{1}{n} [\text{diag}(e^u)\mathbf{B}^\top] \frac{s}{\mathbf{B}e^u}
+    ,
+
+and the Hessian reads
+
+.. math::
+    :label: raw-hessian
+
+    \nabla^2 F(u) = \frac{1}{n} \text{diag}(e^u) \text{diag}(\mathbf{B}^\top \frac{s}{\mathbf{B}e^u}) - \frac{1}{n} \text{diag}(e^u) \mathbf{B}^\top \text{diag}(\frac{s}{(\mathbf{B}e^u)^2})\mathbf{B}\text{diag}(e^u)
+    ,
+
+where the fraction and the square operations are performed element-wise.
+
+The Hessian, as it is, is costly to evaluate because of the right hand-side term. In particular, the latter involves a :math:`\mathcal{O}(n^3)` operations. We overcome this limitation by deriving a diagonal upper bound on the Hessian.
```

Comment:
```suggestion
The Hessian, as it is, is costly to evaluate because of the right hand-side term. In particular, the latter involves a :math:`\mathcal{O}(n^3)` operations. We overcome this limitation by using a diagonal upper bound on the Hessian.
```